### DISCUSSION TITLE
Error: snappy: corrupt input (unsupported chunk length: 0)

### DISCUSSION THREAD

**Discussion by ibrahimnamdar** [datalust/seq-tickets#2325] [Help] [answered]
Hello,

We recently migrated our SEQ logs from a Windows Server to Linux (ubuntu 22.04) Docker container in Azure.
We are getting the following error on SEQ Log Indexing Process:

{"@t":"2024-11-25T23:51:05.3159073Z","@sp":"8505a9f7897d7c4d","@tr":"8b6237624825ea9f082059edcc8667cc","@mt":"Indexing with {Allocated:0.000} ms allowance","@r":["600000.000"],"@l":"Error","@x":"Flare.Ffi.Result.FlareException: Flare native storage failed (IOError), the task encountered an error in a worker.\n   caused by: error evaluating the next record to index\n   caused by: snappy: error snappy reading frame header\n   caused by: snappy: corrupt input (unsupported chunk length: 0)\n   at Seq.Engine.Storage.StorageEngine.QueryInternal(Cursor cursor, NativeCancellationToken nativeCancel, Object[] sharedColumnBuffer, CancellationToken cancel)+MoveNext()\n   at Seq.Engine.Events.EventStore.QueryInternal(IEnumerable`1 rowset, Boolean disableReadRateLimit)+MoveNext()\n   at System.Linq.Enumerable.TryGetSingle[TSource](IEnumerable`1 source, Boolean& found)\n   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)\n   at Seq.Engine.Queries.DataStore.WriteNextIndexSet(CancellationToken cancel)\n   at Seq.Server.Features.Indexing.EventStoreIndexer.Index(TimeSpan allocatedIndexingTime, CancellationToken cancel)","Allocated":600000,"@st":"2024-11-25T23:50:58.0895946Z","SourceContext":"Seq.Server.Features.Indexing.EventStoreIndexer"}

We tried the `flaretl repair --metadata /data/Stream` command but no luck.

Any ideas?

Kind regards,
Ibrahim

**Comment by nblumhardt** [comment]
Hi Ibrahim, thanks for your message.

If the new install is hosted in an Azure Web App, or using the non-premium Azure Files backing storage, this issue will likely continue to appear; let me know if that's the case.

Because this error is in the stored event data, rather than metadata, you'll need to drop the `--metadata` flag from the `flaretl repair` command-line.

I'm not 100% certain the repair process will cover this case, but if it produces output that should at least identify the problem chunk, and at that point we should be able to provide further steps to sort it out.

**Comment by ibrahimnamdar** [comment]
Hi Nicholas,

Thank you for the answer.

The new install is on a Linux (ubuntu 22.04) VM with Premium SSD v2 LRS disks attached.

Here are the results of the commands:

1. `flaretl repair /data/Stream`
`No storage problems were found`
2. `flaretl repair /data/Stream/stream.flare`
`No storage problems were found`

It seems like flaretl repair didn't find anything. 
Anything else I can try?

**Reply by nblumhardt** [reply]
Thanks for the update. That's interesting........

One thing that might help narrow the issue down: 

Open Events, don't select any signals, and set the time range to "All time", run:

```
select count(*) from stream where @Message like '%test%'
```

this should force a decompressed read of every record (and hopefully not time out).

By repeatedly halving the range and re-running the query, you should then be able to narrow down the failure to a short time period, and with that we might be able to find an alternative path forward.

**Reply by ibrahimnamdar** [reply]
Sorry it took a while to find the failure, we have a lot of logs. I was able to narrow the failure down to a 5min period:

![image](https://github.com/user-attachments/assets/17b20254-e882-4ab1-909b-7898d44e5418)

**Reply by KodrAus** [reply]
Thanks for narrowing that down @ibrahimnamdar. Would it be possible to grab an `ls` of your `/data/Stream` directory? I think our path forward will be to identify the offending `.span` file, remove it, and re-run the `flaretl repair` command. We should be able to work out which one it is through the file listing.

**Reply by ibrahimnamdar** [reply]
Hello,

The token value in the screenshot keeps changing every time I run the query so I couldn't really match it to any file in the Stream folder.
Here is the `ls` result:

[Stream Folder LS.txt](https://github.com/user-attachments/files/17940406/Stream.Folder.LS.txt)

**Reply by KodrAus** [reply]
Thanks @ibrahimnamdar. That's ok, we can work out which file covers that range and get back to you :+1:

**Reply by ibrahimnamdar** [reply]
Thank you @KodrAus
I'll be waiting for your response

**Reply by KodrAus** [reply]
Thanks @ibrahimnamdar. One last thing I need is your UTC offset. Those file names encode UTC timestamp ranges, which when decoded look like this:

[stream-ls.txt](https://github.com/user-attachments/files/17952687/stream-ls.txt)

The offending file is the one(s) that overlap the time range from your screenshot, which will be in your local offset.

**Reply by ibrahimnamdar** [reply]
Thanks for decoding @KodrAus
My local timezone is UTC +1. Do I delete the file that corresponds to the failure time?

**Reply by nblumhardt** [reply, marked as answer]
Hi @ibrahimnamdar;

Moving it to an offline location would be safer than deleting, but yes :+1: - here's the process:

1. Stop the Seq container (important)
2. Move the problem `.span` file to a separate storage location
3. Either through an init script, or by mounting the storage volume to a temporary container instance invoked with the `shell` argument*, run `flaretl repair --metadata /data/Stream/stream.flare`
4. Start the Seq container

(* not sure you'll have encountered this, but it would look something like `docker run --rm -it -v (host data path):/data datalust/seq shell`)

**Reply by ibrahimnamdar** [reply]
Thank you @nblumhardt and @KodrAus 🙏,

I moved the problem file out of the `Stream` folder and ran `flaretl repair /data/Stream` cmd via an init script.
Indexing was successful after the script ran.

**Reply by KodrAus** [reply]
Thanks @ibrahimnamdar. For what it’s worth, we’ve since improved these error messages to include the path of the file they encountered a corruption in to make them easier to identify in future releases. This is extremely uncommon though, especially on the kind of file you found one in, so hopefully won’t come up again.

**Comment by ibrahimnamdar** [comment]
Yeah, not sure how the corruption happened but it probably has to do with our log file migration.
Seeing the corrupted file name in SEQ internal logs would definitely lead me to move the file and see what happens (on a test VM of course). 

Thanks for the improvement and support
